Домашнее задание №3. Первая попытка сдачи.

IgushkinV opened the pull request.

IgushkinV committed:
Домашнее задание №3. Первая попытка сдачи.

tlt-java-courses commented on src/main/java/Igushkin/Lesson3/Address.java:
```diff
@@ -0,0 +1,81 @@
+package Igushkin.Lesson3;
```
Почему в этом пул реквесте фигурирует какой-то файл из пакета с lesson1?

IgushkinV replied:
Это потому, что изначально шаблон ветки master был с папкой Lesson1. Он удален.

tlt-java-courses commented on src/main/java/Igushkin/Lesson3/HWMethods.java:
```diff
@@ -0,0 +1,129 @@
+     */
+    public void findDuplicates(ArrayList<Human> humans) {
+        System.out.println("Дублированные элементы в переданном ArrayList: ");
+        for (int i = 0; i < humans.size(); i++) {
```
Окей, а если использовать для этого дела коллекции?

IgushkinV replied:
Переделал.

IgushkinV committed:
Домашнее задание №3. Переделал метод нахождения повторяющихся элеметов, используя HashMap.

IgushkinV committed:
Альтернативная реализация последних двух заданий с сортировкой HashMap по ключу и по значению с небольшим дополнением.
Класс MultiSortingMap хранит коллекцию Map<K,V> и позволяет сортировать коллецию по ключу, по значению, по порядку добавления элементов,
 добавлять и удалять элементы, получать первый и последний элементы в зависимости от режима сортировки, получать
 самый старый и самый новый элементы.
 Класс SortingMode является перечислением enum и хранит названия режимов сортировки для класса MultiSortingMap.

IgushkinV committed:
Убран лишний код из SortingMode
